core/src/main/java/com/sequenceiq/cloudbreak/converter/RecipeRequestToRecipeConverter.java

```diff
@@ -49,6 +77,23 @@ public Recipe convert(RecipeRequest json) {
         return recipe;
     }
 
+    private void setOrganization(RecipeRequest json, Recipe recipe) {
+        IdentityUser identityUser = authenticatedUserService.getCbUser();
+        try {
+            transactionService.required(() -> {
+                Set<Organization> usersOrganizations = organizationService.retrieveForUser(identityUser);
+                Organization organization = organizationService.get(json.getOrganizationId());
+                if (!usersOrganizations.contains(organization)) {
+                    throw new NotFoundException("Organization not found for user.");
```

Comment:
Maybe something like this would be better: String.format("User is not part of the %s organization", org.name)

Reply:
If the sent organization ID is for a completely other tenant we would leak an organization name of a different company. I could make a check if the org is in the user's tenant and only then include the org name in the exception, but I'm not sure it's worth the hassle